fleetspeak/src/client/https/https.go

```diff
@@ -81,7 +81,7 @@ func makeTransport(cctx comms.Context, dc func(ctx context.Context, network, add
 	}
 
 	return ci.ID, &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
+		Proxy: http.ProxyURL(si.Proxy),
```

Comment:
It may make sense to keep ProxyFromEnvironment as a fallback mechanism, in case the proxy is not specified in the config.

Reply:
Done.

I've tried to introduce a test for the ProxyFromEnvironment case, but it had some problems:

- ProxyFromEnvironment caches the proxy setting on first use, so I can't override it in the test. 
- ProxyFromEnvironment wont' use a proxy for requests to localhost.

Reply:
Added a TODO in the code as discussed.